projects/memcached-clusters/cache-tester/cmd/root.go

```diff
@@ -0,0 +1,69 @@
+package cmd
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/bradfitz/gomemcache/memcache"
+)
+
+func Execute() {
+	mcrouter := flag.String("mcrouter", "", "port to connect to mcrouter")
+	memcacheds := flag.String("memcacheds", "", "comma separated list of memcached servers ports")
+	flag.Parse()
+
+	if *mcrouter == "" || *memcacheds == "" {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	// Connect to mcrouter
+	mcr := memcache.New("localhost:" + *mcrouter)
+	defer mcr.Close()
+
+	// Connect to memcached servers
+	ports := strings.Split(*memcacheds, ",")
+	mcs := make([]*memcache.Client, 0)
+	for i, port := range ports {
+		mcs = append(mcs, memcache.New("localhost:"+port))
+		defer mcs[i].Close()
+	}
+
+	// Set a key in mcrouter
+	err := mcr.Set(&memcache.Item{Key: "hello", Value: []byte("world")})
+	if err != nil {
+		fmt.Printf("Error setting key in mcrouter: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Get the key from each memcached server
+	items := make([]*memcache.Item, 0)
+	item := new(memcache.Item)
+	for i, mc := range mcs {
+		item, err = mc.Get("hello")
+		if err != nil {
+			fmt.Printf("Error getting key from memcached server %v: %v\n", i, err)
+		}
+		if err == memcache.ErrCacheMiss {
```

Comment:
This is probably a good enough check (because we assume the cluster is _either_ sharding _or_ replicating, and that things work), but if I was doing this in a system that may change over time, or which I didn't trust so much, I'd probably want to check that exactly one replica returned a value, and actually have three possible outcomes from the check: sharded, replicated, or unknown.